src/main/java/net/fabricmc/loom/AbstractPlugin.java

```diff
@@ -339,7 +339,7 @@ protected void configureCompile() {
 				extension.addUnmappedMod(jarTask.getArchivePath().toPath());
 				remapJarTask.getAddNestedDependencies().set(true);
 
-				remapJarTask.doLast(task -> project1.getArtifacts().add("archives", remapJarTask.getArchivePath()));
+				project1.getArtifacts().add("archives", remapJarTask.getArchivePath());
```

Comment:
```suggestion
				project1.getArtifacts().add("archives", remapJarTask);
```
Unrelated to the scope of this PR, but we might as well do it while we're changing this. [See what ArtifactHandler does when adding AbstractArchiveTasks](https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/dsl/ArtifactHandler.html).

Reply:
Sure, I've incorporated this change -- also modified the line below to not fetch the jar task a second time while I was looking at the area.